certifier-osv.md

```diff
@@ -0,0 +1,197 @@
+
+## Architecture
+
+```
+┌──────────────┐     ┌───────────────┐     ┌────────────┐
+│  SBOM Input  │────>│  OSV Certifier│────>│  OSV API   │
+└──────────────┘     └───────────────┘     └────────────┘
+                            │
+                            ▼
+                    ┌───────────────┐
+                    │  GUAC Graph   │
+                    └───────────────┘
+```
+
+## Key Features
+
+- **Vulnerability Detection**: Scans dependencies using SBOMs and cross-references with OSV's vulnerability catalog
+- **Automated Updates**: Regular synchronization with OSV for current vulnerability data
+- **Comprehensive Reporting**: Structured reports showing vulnerabilities by dependency and severity
+- **SBOM Format Support**: 
+  - [CycloneDX](https://cyclonedx.org) (JSON, XML)
+  - [SPDX](https://spdx.dev) (JSON, YAML)
+
+## Integration Details
+
+### Data Collection Process
+
+1. **Dependency Matching**: 
+   - Parses SBOM files for package information
+   - Extracts package names, versions, and ecosystem identifiers
+   - Validates package metadata format
+
+2. **API Querying**: 
+   - Queries OSV using standardized package identifiers
+   - Batch processing for multiple dependencies
+   - Handles API rate limiting and retries
+
+3. **Data Correlation**: 
+   - Maps vulnerabilities to GUAC's dependency graph
+   - Associates vulnerability data with package versions
+   - Maintains relationship between dependencies
+
+### Supported Package Ecosystems
+
+The OSV Certifier supports vulnerability detection across these verified ecosystems:
+
+| Ecosystem | Identifier Format | Example |
+|-----------|------------------|---------|
+| npm | `pkg:npm/{name}@{version}` | `pkg:npm/lodash@4.17.21` |
+| PyPI | `pkg:pypi/{name}@{version}` | `pkg:pypi/requests@2.28.1` |
+| Maven | `pkg:maven/{group}/{artifact}@{version}` | `pkg:maven/org.apache.logging.log4j/log4j-core@2.14.1` |
+| Go | `pkg:golang/{path}@{version}` | `pkg:golang/golang.org/x/text@v0.3.7` |
+| Cargo | `pkg:cargo/{name}@{version}` | `pkg:cargo/serde@1.0.152` |
+| NuGet | `pkg:nuget/{name}@{version}` | `pkg:nuget/Newtonsoft.Json@13.0.1` |
+
+### Feature Support
+
+#### Supported
+- Public vulnerabilities from OSV's database
+- Dependency version mapping with known vulnerabilities
+- Version range analysis
+- [PURL](https://github.com/package-url/purl-spec) package identification
+- Severity classification using [CVSS](https://www.first.org/cvss/) scores
+
+#### Unsupported
+- Private vulnerability detection
+- Ecosystems not covered by OSV
+- Binary vulnerability scanning
+- Custom vulnerability feeds
+
+### Usage
+
+Basic command syntax:
+```bash
+guac osv-certify --sbom <path> [options]
+```
+
+Available options:
+```bash
+Options:
```

Comment:
these options will be:
```
Usage:
  guacone certifier osv [flags]

Flags:
      --certifier-batch-size int   sets the batch size for pagination query for the certifier (default 60000)
      --certifier-latency string   sets artificial latency on the certifier. Defaults to empty string (not enabled) but can set m, h, s...etc
  -h, --help                       help for osv
  -l, --last-scan int              hours since the last scan was run. If not set, run on all packages/sources (default 4)

Global Flags:
      --add-license-on-ingest   if enabled, the ingestor will query and ingest clearly defined for licenses. Warning: This will increase ingestion times
      --add-vuln-on-ingest      if enabled, the ingestor will query and ingest OSV for vulnerabilities. Warning: This will increase ingestion times
      --csub-addr string        address to connect to collect-sub service (default "localhost:2782")
      --csub-tls                enable tls connection to the server
      --csub-tls-skip-verify    skip verifying server certificate (for self-signed certificates for example)
      --gql-addr string         endpoint used to connect to graphQL server (default "http://localhost:8080/query")
      --header-file string      a text file containing HTTP headers to send to the GQL server, in RFC 822 format
  -i, --interval string         if polling set interval, m, h, s, etc. (default "5m")
  -p, --poll                    sets the collector or certifier to polling mode
  ```

Reply:
## Available Options

### Usage
Basic command syntax:
```bash
guacone certifier osv --sbom <path> [options]
```

### Flags
```bash
Flags:
      --certifier-batch-size int   sets the batch size for pagination query for the certifier (default 60000)
      --certifier-latency string   sets artificial latency on the certifier. Defaults to empty string (not enabled) but can set m, h, s...etc
  -h, --help                       help for osv
  -l, --last-scan int              hours since the last scan was run. If not set, run on all packages/sources (default 4)
```
### Global Flags:
```bash
      --add-license-on-ingest   if enabled, the ingestor will query and ingest clearly defined for licenses. Warning: This will increase ingestion times
      --add-vuln-on-ingest      if enabled, the ingestor will query and ingest OSV for vulnerabilities. Warning: This will increase ingestion times
      --csub-addr string        address to connect to collect-sub service (default "localhost:2782")
      --csub-tls                enable tls connection to the server
      --csub-tls-skip-verify    skip verifying server certificate (for self-signed certificates for example)
      --gql-addr string         endpoint used to connect to graphQL server (default "http://localhost:8080/query")
      --header-file string      a text file containing HTTP headers to send to the GQL server, in RFC 822 format
  -i, --interval string         if polling set interval, m, h, s, etc. (default "5m")
  -p, --poll                    sets the collector or certifier to polling mode
```